code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1863,29 +1863,31 @@
 
 //Works out blood pressure alert level -- not very accurate
 /mob/living/carbon/human/proc/get_blood_pressure_alert()
-	var/list/bp = blood_pressure()
+	var/list/bp_list = blood_pressure()
 	// For a blood pressure, e.g. 120/80
 	var/systolic_alert // this is the top number '120' -- highest pressure when heart beats
 	var/diastolic_alert // this is the bottom number '80' -- lowest pressure when heart relaxes
-
-	switch(bp[1])
-		if(BP_HIGH_SYSTOLIC to INFINITY)
+	
+	var/blood_pressure_systolic = bp_list[1]
+	if (blood_pressure_systolic)
+		if (blood_pressure_systolic >= get_bp_high_systolic())
 			systolic_alert = BLOOD_PRESSURE_HIGH
-		if(BP_PRE_HIGH_SYSTOLIC to BP_HIGH_SYSTOLIC)
+		else if (blood_pressure_systolic >= get_pre_high_systolic() && blood_pressure_systolic <= get_bp_high_systolic())
 			systolic_alert = BLOOD_PRESSURE_PRE_HIGH
-		if(BP_IDEAL_SYSTOLIC to BP_PRE_HIGH_SYSTOLIC)
+		else if (blood_pressure_systolic >= get_ideal_systolic() && blood_pressure_systolic <= get_pre_high_systolic())
 			systolic_alert = BLOOD_PRESSURE_IDEAL
-		if(-INFINITY to BP_IDEAL_SYSTOLIC)
+		else if (blood_pressure_systolic <= get_ideal_systolic())
 			systolic_alert = BLOOD_PRESSURE_LOW
 
-	switch(bp[2])
-		if(BP_HIGH_DIASTOLIC to INFINITY)
+	var/blood_pressure_disatolic = bp_list[2]
+	if (blood_pressure_disatolic)
+		if (blood_pressure_disatolic >= get_bp_high_disatolic())
 			diastolic_alert = BLOOD_PRESSURE_HIGH
-		if(BP_PRE_HIGH_DIASTOLIC to BP_HIGH_DIASTOLIC)
+		else if (blood_pressure_disatolic >= get_pre_high_disatolic() && blood_pressure_disatolic <= get_bp_high_disatolic())
 			diastolic_alert = BLOOD_PRESSURE_PRE_HIGH
-		if(BP_IDEAL_DIASTOLIC to BP_PRE_HIGH_DIASTOLIC)
+		else if(blood_pressure_disatolic >= get_ideal_disatolic() && blood_pressure_disatolic <=  get_pre_high_disatolic())
```

Comment:
you can save a lit of processing by having the "ideal" rate be the first thing to be checked, one factor off ideal being the second, and then the high one being the last. not very important, but just handy to know